src/sources/ukt.rs

```diff
@@ -0,0 +1,363 @@
+	}
+}
+
+impl UktSource {
+	/// Create a new UKT source
+	pub fn new(ukt_config: UktSourceConfig) -> Self {
+		let client = Client::new();
+
+		Self { ukt_config, client }
+	}
+
+	/// Get list of user emails that have been removed
+	pub async fn get_removed_user_emails(&self) -> Result<Vec<String>> {
+		let oauth2_token = self.get_oauth2_token().await?;
+		let email_list = self.fetch_list(oauth2_token).await?;
+
+		Ok(email_list)
+	}
+
+	/// Get the OAuth2 token
+	async fn get_oauth2_token(&self) -> Result<OAuth2Token> {
+		let mut params = HashMap::new();
+		params.insert("grant_type", &self.ukt_config.grant_type);
+		params.insert("scope", &self.ukt_config.scope);
+		params.insert("client_id", &self.ukt_config.client_id);
+		params.insert("client_secret", &self.ukt_config.client_secret);
+
+		let response =
+			self.client.post(self.ukt_config.oauth2_url.clone()).form(&params).send().await?;
+
+		response.error_for_status_ref().context("UKT oAuth2 received non-OK status code")?;
+
+		let response: serde_json::Value = response.json().await?;
+
+		if let Some(error) = response.get("error") {
+			anyhow::bail!("Error in UKT oAuth2 response body: {}", error)
+		}
+
+		let oauth2_token: OAuth2Token = serde_json::from_value(response)
+			.context("Failed to deserialize oAuth2 token response")?;
+
+		Ok(oauth2_token)
+	}
+
+	/// Fetch the list of users
+	async fn fetch_list(&self, oauth2_token: OAuth2Token) -> Result<EmailList> {
+		let current_date = Utc::now().format("%Y%m%d").to_string();
+
+		let response = self
+			.client
+			.get(self.ukt_config.endpoint_url.clone())
+			.query(&[("date", &current_date)])
+			.bearer_auth(oauth2_token.access_token)
+			.header("x-participant-token", oauth2_token.id_token)
+			.send()
+			.await?;
+
+		response.error_for_status_ref().context("UKT endpoint received non-OK status code")?;
+
+		let response: serde_json::Value = response.json().await?;
+
+		if let Some(error) = response.get("error") {
+			anyhow::bail!("Error in UKT endpoint response body: {}", error)
+		}
+
+		let email_list: EmailList = serde_json::from_value(response)
+			.context("Failed to deserialize email list response")?;
+
+		Ok(email_list)
+	}
+}
+
+/// List of emails
+type EmailList = Vec<String>;
+
+/// OAuth2 token response
+#[derive(Debug, Deserialize)]
+struct OAuth2Token {
+	/// Access token
+	access_token: String,
+	/// ID token
+	id_token: String,
+}
+
+/// Configuration to get a list of users from UKT
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+pub struct UktSourceConfig {
+	/// The URL of the endpoint provided by UKT
+	pub endpoint_url: Url,
+	/// The OAuth2 URL
+	pub oauth2_url: Url,
+	/// The API client ID
+	pub client_id: String,
+	/// The API client secret
+	pub client_secret: String,
+	/// The scope
+	pub scope: String,
+	/// The grant type
+	pub grant_type: String,
+}
+
+/// Helper module for unit and e2e tests
+pub mod test_helpers {
+	#![allow(clippy::expect_used, clippy::unwrap_used)]
```

Comment:
```rust
#[cfg(test)]
pub mod test_helpers {
...
```
with 
```toml
allow-expect-in-tests = true
allow-unwrap-in-tests = true
```
in `.clippy.toml` (see [.clippy.toml](https://github.com/famedly/backend-system-tests/blob/ef787dc3d6a882e18f1333abf9c5ee73e7935151/.clippy.toml))

Reply:
Didn't realize there were exception options specific to tests, maybe we should do that in a separate PR to also bring up the rest of the tests up to speed.

Reply:
Okay.